docs/okeps/okep-5296-bgp.md

```diff
@@ -0,0 +1,663 @@
+which in which case OVN-Kubernetes will advertise routes in the VRF that corresponds to the selected network. Any other
+values other than "auto" or "default" will result in an error. There is no support at this time for routing leaking into
+any other VRF.
+
+The frrConfigurationSelector is used in order to determine which FRRConfiguration CR to use for building the OVN-Kubernetes
+driven FRRConfiguration. OVN-Kubernetes needs to leverage a pre-existing FRRConfiguration to be able to find required
+pieces of configuration like BGP peering, etc. If more than one FRRConfiguration is found matching the selector, then
+an error will be propagated to the RouteAdvertisements CR and no configuration shall be done.
+
+#### No Tunnel/Overlay Mode
+
+Changes will be made to the CUDN and UDN CRDs in order to specify the method of transport for east/west traffic. A new
+field, “transport” will be added with values “geneve” or “none” (default geneve). The transport option in the CRD may be
+used to toggle between using Geneve encapsulation (the default today) or using no encapsulation. With no encapsulation,
+east/west packets are routed directly on the underlay using routing learned via BGP. This is supported for Layer 3
+networks only. Enabling a transport of “none” for Layer 2 networks will result in an error status in the network CR.
+Enabling a transport of “none” without selecting all nodes for RouteAdvertisements will also break east/west traffic
+between those nodes, as there will not be routes propagated into the BGP network for unselected nodes. Users should
+expect disruption when the transport of a network is changed.
+
+### Implementation Details
+
+#### BGP Configuration
+
+When OVN-Kubernetes detects that a FRRConfiguration has been created that has a corresponding and valid FRRNodeState,
+OVN-Kubernetes will then use RouteAdvertisements CR create a corresponding FRRConfiguration. The following examples will
+use an environment where a user has created an FRRConfiguration:
+
+```yaml
+apiVersion: frrk8s.metallb.io/v1beta1
+kind: FRRConfiguration
+metadata:
+  creationTimestamp: "2024-06-11T14:22:37Z"
+  generation: 1
+  name: metallb-ovn-worker
+  namespace: metallb-system
+  resourceVersion: "1323"
+  uid: 99b64be3-4f36-4e0b-8704-75aa5182d89f
+  labels:
+    routeAdvertisements: default
+spec:
+  bgp:
+    routers:
+    - asn: 64512
+      neighbors:
+      - address: 172.18.0.5
+        asn: 64512
+        disableMP: false
+        holdTime: 1m30s
+        keepaliveTime: 30s
+        passwordSecret: {}
+        port: 179
+        toAdvertise:
+          allowed:
+            mode: filtered
+        toReceive:
+          allowed:
+            mode: filtered
+  nodeSelector:
+    matchLabels:
+       kubernetes.io/hostname: ovn-worker
+```
+
+OVNKube-Controller will check that if this FRRConfiguration applies to its node, ovn-worker. For this example, a user-defined
```

Comment:
nit for CNCF correctness here and every to use either `OVN-Kubernetes` or `ovn-kubernetes` for project name and 
`ovnkube-controller`? or `OVN-Kubernetes Controller`